main.py

```diff
@@ -33,42 +47,96 @@ def main():
 
     for pr in prs:
         print(f"\nClassifying files from PR {pr} for predictions training ")
-        processing.process_files(
-            ai, db, pr
-        )  # Here is where ASTs and classification is done. All the "heavy lifting" of the core engine
-
-    db.save()
-    db.close()
-
-    # Pseudo code below... just the concepts (and thoughts, this is not set in stone... what do you think?)
-    # import llm_classifier
-    # training_model == (get from command line)
-    #
-    # if training_model == "GPT":
-
-    #     # Generate fine tuning file
-    #
-    #     data_from_extraction = database manager query OutputTable
-    #
-    #     system_message, assistant_message = llm_classifier.generate_system_message(domain_dictionary, data_from_extraction)
-    #     generate_gpt_messages(system_message, assistant_message, df)
-
-    #     # Fine tune GPT Model
-    #     llm_classifier = fine_tune_gpt(openAI_key)
-    #     save_model(llm_classifier)
 
-    # if model == "Random Forest":
+        # Here is where ASTs and classification are done;
+        # all the "heavy lifting" of the core engine
+        processing.process_files(ai, db, pr)
 
-    #     data_from_extraction = database manager query OutputTable
-    #
-    #     rf_model = llm_classifier.train_rf(domain_dictionary, data_from_extraction)
-
-    #     save_model(rf_model)
+    db.save()
 
-    # exit()
+    method = cfg_dict["classification_method"]
+
+    print("\nPreparing data frame")
+    df = get_prs_df(db, prs)
+
+    if method == "gpt":
+        system_message, assistant_message = classifier.generate_system_message(
+            sub_labels, df
+        )
+        classifier.generate_gpt_messages(system_message, assistant_message, df)
+
+        llm_classifier = classifier.fine_tune_gpt()
+
+        # Save Model
+        with open(cfg_dict["classification_model_save"], "wb") as f:
```

Comment:
`classification_model_save` seems to be a path where a classification model will be output to. I had to look around to understand what it meant and I think a more fitting name could be chosen.

The most important part imo is that 'save' is typically read as a verb, not a noun that describes a location. Because of this, this initially seemed to me like the variable name for a boolean. Something like `classification_model_path` or `classification_model_output_path` would be better. 

If we want to be shorter, scikit-learn commonly uses `clf` as a shorthand for the concept of a classifier (whether this is the same type of classifier may or may not be important; see [here](https://scikit-learn.org/stable/tutorial/basic/tutorial.html#learning-and-predicting)). So, something like `clf_model_path`/`clf_model_out_path` might be clear to those in this niche of CS while also being shorter. The downside is that it might confuse anyone with sklearn experience.

```suggestion
        with open(cfg_dict["clf_model_out_path"], "wb") as f:
```